### ISSUE TITLE
Add functionality for deleting adapter layers in PEFT integration

### ISSUE THREAD

**Issue by itsskofficial** [huggingface/transformers#34649]
### Feature request

This request aims to introduce functionality to delete specific adapter layers integrated with PEFT (Parameter-Efficient Fine-Tuning) within the Hugging Face Transformers library. This would enable users to manage memory and computational resources more efficiently by unloading adapters that are no longer needed during model inference or fine-tuning.

### Motivation

This feature request addresses scenarios where users load multiple adapters during fine-tuning or inference but need the ability to selectively unload adapters without reloading the entire model. This enhancement is crucial for optimizing performance in memory-constrained environments and enhancing the flexibility of adapter management within Transformer models.

### Your contribution

Yes, I am submitting a PR

**Comment by LysandreJik** [comment]
cc @benjaminbossan, WDYT?

**Comment by BenjaminBossan** [comment]
Yes, makes sense, I reviewed the PR. See also [this previous comment](https://github.com/huggingface/transformers/issues/32933#issuecomment-2379522352).
